packages/instrumentor/instrument.ts

```diff
@@ -110,20 +122,39 @@ export function shouldInstrumentFn(
 	};
 }
 
-function instrument(
+export function instrument(
 	code: string,
 	filename: string,
-	shouldInstrument: FilePredicate,
-	shouldHook: FilePredicate
+	options: InstrumentorOptions
 ) {
 	const transformations: PluginItem[] = [];
-	if (shouldInstrument(filename)) {
-		transformations.push(codeCoverage, compareHooks);
+
+	const shouldInstrument = shouldInstrumentFn(
```

Comment:
`shouldInstrument` and `shouldHook` are created for every file. If they should stay here, we could remove one level of indirection and directly call the functions with all needed arguments and so get rid of the higher-order function `shouldInstrumentFn` and `bind` on `hasFunctionsToHook`. Another option would be to pass in the functions in the `InstrumentorOptions`. I think I would like that more.

Reply:
As discussed, I've refactored the instrumentor code into a dedicated class.

Reply:
:+1: